test/JsonApiDotNetCoreMongoDbExampleTests/IntegrationTests/Sorting/SortTests.cs

```diff
@@ -0,0 +1,202 @@
+            {
+                Caption = "X"
+            };
+
+            await _testContext.RunOnDatabaseAsync(async db =>
+            {
+                await db.GetCollection<Article>(nameof(Article)).InsertOneAsync(article);
+            });
+
+            var route = $"/api/v1/articles/{article.StringId}/author?sort=id";
+
+            // Act
+            var (httpResponse, responseDocument) = await _testContext.ExecuteGetAsync<ErrorDocument>(route);
+
+            // Assert
+            httpResponse.Should().HaveStatusCode(HttpStatusCode.BadRequest);
+
+            responseDocument.Errors.Should().HaveCount(1);
+            responseDocument.Errors[0].StatusCode.Should().Be(HttpStatusCode.BadRequest);
+            responseDocument.Errors[0].Title.Should().Be("The specified sort is invalid.");
+            responseDocument.Errors[0].Detail.Should().Be("This query string parameter can only be used on a collection of resources (not on a single resource).");
+            responseDocument.Errors[0].Source.Parameter.Should().Be("sort");
+        }
+
+        [Fact]
+        public async Task Cannot_sort_on_attribute_with_blocked_capability()
+        {
+            // Arrange
+            var route = "/api/v1/todoItems?sort=achievedDate";
+
+            // Act
+            var (httpResponse, responseDocument) = await _testContext.ExecuteGetAsync<ErrorDocument>(route);
+
+            // Assert
+            httpResponse.Should().HaveStatusCode(HttpStatusCode.BadRequest);
+            
+            responseDocument.Errors.Should().HaveCount(1);
+            responseDocument.Errors[0].StatusCode.Should().Be(HttpStatusCode.BadRequest);
+            responseDocument.Errors[0].Title.Should().Be("Sorting on the requested attribute is not allowed.");
+            responseDocument.Errors[0].Detail.Should().Be("Sorting on attribute 'achievedDate' is not allowed.");
+            responseDocument.Errors[0].Source.Parameter.Should().Be("sort");
+        }
+
+        [Fact]
+        public async Task Can_sort_descending_by_ID()
+        {
+            // Arrange
+            var persons = new List<Person>
+            {
+                new Person {LastName = "B"},
+                new Person {LastName = "A"},
+                new Person {LastName = "A"}
+            };
+
+            await _testContext.RunOnDatabaseAsync(async db =>
+            {
+                var collection = db.GetCollection<Person>(nameof(Person));
+                await collection.DeleteManyAsync(Builders<Person>.Filter.Empty);
+                await collection.InsertManyAsync(persons);
+            });
+
+            var route = "/api/v1/people?sort=lastName,-id";
+
+            // Act
+            var (httpResponse, responseDocument) = await _testContext.ExecuteGetAsync<Document>(route);
+
+            // Assert
+            httpResponse.Should().HaveStatusCode(HttpStatusCode.OK);
+
+            persons.Sort((a, b) => a.LastName.CompareTo(b.LastName) + b.Id.CompareTo(a.Id));
```

Comment:
Instead of this, take the original test and set hardcoded object IDs (you can get some by calling `ObjectId.GenerateNewId()`). The test needs to show it actually performs the sorting, irrespective of the order of insertion. And show it applies nested sorting.

This applies to test `Sorts_by_ID_if_none_specified` too.

Reply:
I did do this but I want to point out that sorting by ID when working with MongoDB will always like sorting by order of insertion since an `ObjectId` is generated based on the current timestamp, a random number and an auto-incrementing counter. Here is the official reference: https://docs.mongodb.com/manual/reference/method/ObjectId/

It is for this reason that in `Can_sort_descending_by_ID` you'll notice I first generate the three ids sequentially and then assign them in reverse order to the `Persons`.

Reply:
> I did do this but I want to point out that sorting by ID when working with MongoDB will always like sorting by order of insertion since an ObjectId is generated based on...

I'm aware of that. And it is the reason I'd like to use _hardcoded_ IDs here. This emulates an API client sending us explicit IDs instead of the server auto-generating them, so they can be in a different order.

To clarify: don't call `ObjectId.GenerateNewId()` in the test code, but execute it in your terminal and copy/paste the resulting values.

Reply:
Oh, and `persons` should be `people`.

Reply:
I don't see any changes.